dms/models/mixins_thumbnail.py

```diff
@@ -14,79 +14,86 @@ class Thumbnail(models.AbstractModel):
     _name = "dms.mixins.thumbnail"
     _description = "Thumbnail Mixin"
 
-    custom_thumbnail = fields.Image(
-        string="Custom Thumbnail",
-        max_width=2048,
-        max_height=2048,
-        attachment=False,
-        prefetch=False,
-    )
-    custom_thumbnail_medium = fields.Image(
+    custom_thumbnail = fields.Binary(string="Custom Thumbnail", attachment=False)
+    custom_thumbnail_medium = fields.Binary(
         "Medium Custom Thumbnail",
-        related="custom_thumbnail",
-        max_width=512,
-        max_height=512,
-        prefetch=False,
+        attachment=True,
+        compute="_compute_custom_thumbnail",
+        store=True,
     )
-    custom_thumbnail_small = fields.Image(
+    custom_thumbnail_small = fields.Binary(
         "Small Custom Thumbnail",
-        related="custom_thumbnail",
-        max_width=512,
-        max_height=512,
-        prefetch=False,
-    )
-
-    thumbnail = fields.Image(
-        compute="_compute_thumbnail",
-        string="Thumbnail",
-        max_width=2048,
-        max_height=2048,
-        prefetch=False,
-    )
-
-    thumbnail_medium = fields.Image(
-        "Medium Thumbnail",
-        related="thumbnail",
-        max_width=512,
-        max_height=512,
-        prefetch=False,
-    )
-
-    thumbnail_small = fields.Image(
-        "SmallThumbnail",
-        related="thumbnail",
-        max_width=512,
-        max_height=512,
-        prefetch=False,
+        attachment=True,
+        compute="_compute_custom_thumbnail",
+        store=True,
     )
+    thumbnail = fields.Binary(compute="_compute_thumbnail", string="Thumbnail",)
+    thumbnail_medium = fields.Binary("Medium Thumbnail", compute="_compute_thumbnail")
+    thumbnail_small = fields.Binary("Small Thumbnail", compute="_compute_thumbnail")
+    icon_url = fields.Char("Icon URL", compute="_compute_icon_url")
 
     # ----------------------------------------------------------
     # Helper
     # ----------------------------------------------------------
 
     @api.model
-    def _get_thumbnail_placeholder_image(self, size, name):
-        path = self._get_thumbnail_path(size, name)
+    def _get_thumbnail_placeholder_image(self, name):
+        path = self._get_thumbnail_path(name)
         with open(path, "rb") as image:
             return base64.b64encode(image.read())
 
     @api.model
-    def _get_thumbnail_path(self, size, name):
+    def _get_thumbnail_path(self, name):
         folders = ["static", "lib", "img", "thumbnails"]
         path = get_resource_path("dms", *folders, name)
         if not os.path.isfile(path):
-            path = get_resource_path("dms", *folders, "file_unkown.svg")
+            path = get_resource_path("dms", *folders, "file_unknown.png")
         return path
 
     def _get_thumbnail_placeholder_name(self):
-        return "folder.svg"
+        return "folder.png"
```

Comment:
Why changing this to PNG? SVG should be the standard.

Reply:
Sorry, it's a mistake when apply commits from 12.0.